### ISSUE TITLE
POST /rest/V1/carts/{cartId}/items

### ISSUE THREAD

**Issue by nickmarx12345678** [magento/magento2#7430]
PRECONDITIONS:
INSTALL MAGENTO 2.0.10
USE THE REST API
ADD A PRODUCT TO YOUR CATALOG, USE ITS ID AS AN ITEM ID
CREATE A CART, GET ITS ID, USE AS PARAMETER IN FOLLOWING REQUESTS AS EITHER CART_ID or QUOTE_ID

STEPS TO REPRODUCE
send an http POST request (https://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html)
with json body:
```
{
    "cartItem": {
        "itemId":1,
        "qty": 1
    }
}
```
to

`/rest/V1/carts/{cartId}/items` where cartId is the id of your cart
THE REAL, ACTUAL RESULT
```
{
  "message": "No such entity with %fieldName = %fieldValue",
  "parameters": {
    "fieldName": "cartId",
    "fieldValue": null
  }
}
```
Additionally, sending the same request with the following json body:
```
{
    "cartItem": {
        "itemId":1,
        "cartId":1,
        "qty": 1
    }
}
```
ACTUAL RESULT
```
{
  "message": "Internal Error. Details are available in Magento log file. Report ID: webapi-5678ac6216cf5"
}
```

EXPECTED RESULTS:
I EXPECT THE "ADD ITEM TO CART" ENDPOINT TO ADD AN ITEM TO THE CART, instead of erroring

log excerpt

```
[2015-12-22 01:50:26] main.CRITICAL: exception 'LogicException' with message 'Report ID: webapi-5678ac6216cf5; Message: Property "CartId" does not have corresponding setter in class "Magento\Quote\Api\Data\CartItemInterface".' in /var/www/magento2/lib/internal/Magento/Framework/Webapi/ErrorProcessor.php:194
Stack trace:
#0 /var/www/magento2/lib/internal/Magento/Framework/Webapi/ErrorProcessor.php(139): Magento\Framework\Webapi\ErrorProcessor->_critical(Object(LogicException))
#1 /var/www/magento2/app/code/Magento/Webapi/Controller/Rest.php(163): Magento\Framework\Webapi\ErrorProcessor->maskException(Object(LogicException))
#2 /var/www/magento2/var/generation/Magento/Webapi/Controller/Rest/Interceptor.php(24): Magento\Webapi\Controller\Rest->dispatch(Object(Magento\Framework\App\Request\Http))
#3 /var/www/magento2/lib/internal/Magento/Framework/App/Http.php(115): Magento\Webapi\Controller\Rest\Interceptor->dispatch(Object(Magento\Framework\App\Request\Http))
#4 /var/www/magento2/lib/internal/Magento/Framework/App/Bootstrap.php(258): Magento\Framework\App\Http->launch()
#5 /var/www/magento2/index.php(39): Magento\Framework\App\Bootstrap->run(Object(Magento\Framework\App\Http))
#6 {main} [] []
```

Futher INFO
Sending the following works as expected, adding an item to cart successfully
```
{
    "cartItem": {
        "itemId":1,
        "quoteId":1,
        "qty": 1
    }
}
```

**Comment by nickmarx12345678** [comment]
this is a duplicate of https://github.com/magento/magento2/issues/2782, which was said to be fixed -- it's not on 2.0.10

**Comment by nickmarx12345678** [comment]
probably you need to update your \Magento\Quote\Api\Data\CartItemInterface $cartItem to include a cartId field, and use that instead of quoteId in the quote instantiation.

**Comment by veloraven** [comment]
@nickmarx12345678 thank you for your report.
Please, format this issue according to the [Issue reporting guidelines](https://github.com/magento/magento2/wiki/Issue-reporting-guidelines): with steps to reproduce, actual result and expected result - so that it is easy to see at once what the problem is.

**Comment by nickmarx12345678** [comment]
@veloraven what about this isn't formatted to your liking

**Comment by paales** [comment]
@nickmarx12345678 It should be in the format from https://github.com/magento/magento2/blob/develop/ISSUE_TEMPLATE.md

**Comment by nickmarx12345678** [comment]
@paales @veloraven 

I believe I formatted it correctly the first time, however I've updated to be more verbose. Idk what else you need here. Magento version, reproduce steps, expected, actual, logs, suggested fix.

**Comment by paales** [comment]
@nickmarx12345678 They are a little nitpicky about the format. Easier to just jump through the hoop ;)

**Comment by SerhiyShkolyarenko** [comment]
@nickmarx12345678 thank you! We checked: the fix was delivered to 2.1, but not to 2.0 branch.
Internal issue for backporting is MAGETWO-61912.

**Comment by erikhansen** [comment]
@SerhiyShkolyarenko Can you provide a commit hash for MAGETWO-61912?

**Comment by magento-engcom-team** [comment]
@nickmarx12345678 @erikhansen, thank you for your report.
The fix for this issue is already available in the `2.1`, `2.2` and `develop` branch. See commit https://github.com/magento/magento2/commit/097c8db99d1802e38c4e4e53a3537b423add21bf
